### PR TITLE
feat: Address badges

### PR DESCRIPTION
**Pull request by vbaranov**
Resolves https://github.com/blockscout/blockscout/issues/10677

## Motivation

Implementing address badges functionality.

## Changelog

`badges` array passed to all API endpoints, which return address info:
```
"badges": [
            {
                "category": "scam",
                "content": "test2"
            }
        ]
```

Management API:

Screenshot 2024-09-12 at 10 40 08

[Blockscout API v2 address badges.postman_collection.json](https://github.com/user-attachments/files/16975522/Blockscout.API.v2.address.badges.postman_collection.json)

Badges live examples:
https://eth-sepolia.k8s-dev.blockscout.com/api/v2/addresses/0x2a202d07f520f6aaee68d2dac9b2ef35834d5923
https://eth-sepolia.k8s-dev.blockscout.com/api/v2/transactions/0x6e8adcb2c078e6689236ef7afe3a02d86707815537a05f38598d5eba24a64b8a

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [x] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [x] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/for-developers/information-and-settings/env-variables).
- [x] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [x] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/routers/address_badges_v2_router.ex [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    plug(RateLimit)
+    plug(:fetch_session)
+  end
+
+  scope "/", as: :api_v2 do
+    pipe_through(:api_v2_no_forgery_protect)
+
+    post("/", V2.AddressBadgeController, :create_badge)
+    post("/:badge_id/addresses/add", V2.AddressBadgeController, :add_addresses_to_badge)
+    post("/:badge_id/addresses/remove", V2.AddressBadgeController, :remove_addresses_to_badge)
```
Comment (fedor-ivn) [inline apps/block_scout_web/lib/block_scout_web/routers/address_badges_v2_router.ex:51]: What do you think about such design?

- `POST` `/:badge_id/addresses` - add address to badge
- `DELETE` `/:badge_id/addresses/:address_hash` - remove address from badge
- `GET` `/:badge_id/addresses` - show badge addresses

Such design seems more RESTful to me

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_badge_controller.ex [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        {:error, :badge_creation_failed}
+
+      _ ->
+        {:api_key, :wrong}
+    end
+  end
+
+  def add_addresses_to_badge(_, _), do: {:error, :not_found}
+
+  def remove_addresses_to_badge(
```
Comment (fedor-ivn) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_badge_controller.ex:87]: What about `assign_badge_to_address` / `unassign_badge_from_address`?

File: apps/block_scout_web/lib/block_scout_web/routers/address_badges_v2_router.ex (hunk 1 above)
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/routers/address_badges_v2_router.ex]: Totally agree. Implemented in https://github.com/blockscout/blockscout/pull/10747/commits/7e5d7fec409a020aed7b60fb513c0e03bae086f0.

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_badge_controller.ex (hunk 2 above)
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/address_badge_controller.ex]: Why not. https://github.com/blockscout/blockscout/pull/10747/commits/7e5d7fec409a020aed7b60fb513c0e03bae086f0.

**Pull request closed without merging**